src/shared/tools.h

```diff
@@ -672,6 +673,11 @@ template <class T> struct vector
         add();
     }
 
+    void push_back( T const& el )
```

Comment:
I've missed that method recently in my own upcoming PR, too. In the future, please consider sending separate PRs for those.

Reply:
originally, it was in a distinct commit, but yes, should have done that.